File: test-suite/src/test/java/io/micronaut/jsonschema/test/SchemaGenerationTest.java
package io.micronaut.jsonschema.test;

import io.micronaut.core.io.ResourceLoader;
import io.micronaut.json.JsonMapper;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

import java.io.IOException;
import java.io.InputStream;
import java.nio.charset.StandardCharsets;
import java.util.Map;
import java.util.Optional;

import static org.junit.jupiter.api.Assertions.*;

@MicronautTest(startApplication = false)
class SchemaGenerationTest {
    @Inject
    ResourceLoader resourceLoader;

    @ParameterizedTest
    @ValueSource(strings = {"llama", "bird", "possum", "possum-environment", "red-winged-blackbird", "salamander" })
    void buildJsonSchema(String name) {
        assertTrue(resourceLoader.getResource("META-INF/schemas/" + name + ".schema.json").isPresent());
    }

    @Test
    void llamaSchema(JsonMapper jsonMapper) throws IOException {
        Optional<InputStream> expectedOptional = resourceLoader.getResourceAsStream("expected-llama.schema.json");
        assertTrue(expectedOptional.isPresent());
        String expected = new String(expectedOptional.get().readAllBytes(), StandardCharsets.UTF_8);
        Optional<InputStream> resultOptional = resourceLoader.getResourceAsStream("META-INF/schemas/llama.schema.json");
        assertTrue(resultOptional.isPresent());
        String result = new String(resultOptional.get().readAllBytes(), StandardCharsets.UTF_8);
        assertEquals(jsonMapper.readValue(expected, Map.class), jsonMapper.readValue(result, Map.class));
    }
}

File: test-suite-python/src/test/python/micronaut/jsonschema/test/SchemaGenerationTest.py
import json
from typing import Annotated

from java.lang import String
from java.nio.charset import StandardCharsets
from jakarta.inject import Inject
from micronaut.core.io import ResourceLoader
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test


@MicronautTest(startApplication=False)
class SchemaGenerationTest:
    resource_loader: Annotated[ResourceLoader, Inject]

    @Test
    def test_build_json_schema(self) -> None:
        for name in ["llama", "red-winged-blackbird"]:
            assert self.resource_loader.getResource("META-INF/schemas/" + name + ".schema.json").isPresent()

    @Test
    def test_llama_schema(self) -> None:
        expected = self.read_resource("expected-llama.schema.json")
        result = self.read_resource("META-INF/schemas/llama.schema.json")
        assert json.loads(expected) == json.loads(result)

    def read_resource(self, path: str) -> str:
        stream = self.resource_loader.getResourceAsStream(path)
        assert stream.isPresent()
        return str(String(stream.get().readAllBytes(), StandardCharsets.UTF_8))
